### ISSUE TITLE
0.9.30 Release

### ISSUE THREAD

**Issue by chadyj** [status-im/status-react#6311]
This is an issue to list and discuss remaining issues for the 0.9.30 release.

Remaining issues can be viewed with the `release` tag.
https://github.com/status-im/status-react/labels/release

The 0.9.30 release has several improvements for #cryptolife and Devcon. 

Using this issue to manage the release as team comms are split between Status and Slack
cc @status-im/clojure @rachelhamlin @andytudhope @annadanchenko @lukaszfryc

**Comment by chadyj** [comment]
Conversation about scope from Slack:

> rachel / Dense Immaterial Mantis [17:57]
> + ENS entry point from the profile screen, with updated URL. Assuming the release will happen on the 22nd, we should have the DApp in shape by then.
> + Extensions work from @jeluard
> + HTTP server fix from @andrey - https://github.com/status-im/status-react/issues/6228 
> + Sidechain/RPC connection from @goranjovic  - https://github.com/status-im/status-react/issues/6250 
> 
> Here’s what would be nice, if we can address in the limited time that we have:
> 
> + Ropsten support for Gitcoin trophies—Gitcoin team said they’d deploy to testnet on Friday 12/10
> + Addition of a network list to the HTTP server (no issue yet)
> 
> We will not change the URL for Bounties Network to prague.bounties.network, but we can do a quick test of the DApp. Bounties Network ran smoothly at ETHBerlin so let’s hope for the best.
> 
> Andy:
> 1. Registration DApp: https://github.com/status-im/hackathon-registration-dapp (will be ready to test this week)
> 2. Sidechain with POA/DAIChain integrated into RPC connections (cc @goranjovic @adamb for how theire experiments are going there)
> 3. prague.bounties.network tested (Simona will provide stuff to test against by the 12th)
> 4. Gitcoin kudos integration (Julien, Rachel and I are tracking that, will hopefully have code to test against by the 12th).
> 5. Extensions and JS API working well in Status (@jeluard and @rachel / Dense Immaterial Mantis to discuss that)
> 6. Might need something specific for @divan's work on Whisper visualisation w/ Alethio, but doubt it.
> 

@andytudhope @rachelhamlin Where are we on the move items? If there are issues can you please add the `release` tag or create issues?

The release was supposed to be cut tomorrow so please indicate which ones are must-haves and which ones are optional.

cc @divan @jeluard @goranjovic @adambabik mentioned above ^

**Comment by chadyj** [comment]
0.9.30 releases updated here in the [release notes doc](https://docs.google.com/document/d/1Z0mFlJJkNbRgv6qm80ob5g4PEvsI8cTmbwZpcRPWPho/edit).

**Comment by adambabik** [comment]
Do we have a nightly or a build which has that sidechain configured? I checked the RPC endpoints for some popular methods and they were fine but I don't have a full suite to test everything.

**Comment by rachelhamlin** [comment]
> ENS entry point from the profile screen, with updated URL. Assuming the release will happen on the 22nd, we should have the DApp in shape by then.

In the nightly & tested. 

> HTTP server fix from @Andrey - #6228

This is small. Team is prioritizing extensions work, but this will actually be @alwx and he's planning to take a look later today.

> Extensions work from @jeluard 

Will tag relevant issues; Nastya is testing. 

> Sidechain/RPC connection from @goranjovic - #6250

I'm not sure about this. 

> Ropsten support for Gitcoin trophies—Gitcoin team said they’d deploy to testnet on Friday 12/10

There are some issues with the display of these tokens. If we can't get a PR in today, probably will have to drop it.

> Addition of a network list to the HTTP server (no issue yet)

Not required.

**Comment by chadyj** [comment]
> Do we have a nightly or a build which has that sidechain configured? I checked the RPC endpoints for some popular methods and they were fine but I don't have a full suite to test everything.

I don't believe so. @goranjovic?

**Comment by chadyj** [comment]
Update from release sync:

The team is aiming to cut the release end of day Tuesday Oct 16.  Please try to get your PR's merged before then.

Also thanks in advanced to the team for getting this issues done quickly, reviewing quickly, and testing quickly. Cheers!

**Comment by chadyj** [comment]
Ping @j-zerah @kimjf for release notes

**Comment by goranjovic** [comment]
We have a PR with the heavy lifting needed for the sidechain change (#6260). With it merged, I can post the PR that actually adds those sidechains.

**Comment by yenda** [comment]
Regarding mailservers:

Previously the app was blindly firing requests for all topics. It was using some heuristics to guesstimate when the mailserver was done sending the requests based on how much time has passed since the last expired message arrived (mailserver send expired messages, in the sense that the whisper network would have considered these messages expired and discarded them). 
The problem with that approach is that there is now a timeout on requests and after that timeout expires, the mailserver stops sending messages (to confirm @pilu @adambabik). 
This means that with the current approach, not only does the app tells the user it is fetching messages when it is actually done for up to 10 seconds, but also some of the history might never be fetched.

Recently signals were introduced to notify status react when a request is completed (all messages for the request have been confirmed to be received) or expired (the timeout, 10 sec by default, has been reached before all messages could be sent). I made a couple of PRs that started using these signals instead of the heuristic mentioned in the previous point. However this PR has made a few underlying problems more visible:

- we can only request one topic per request (soon fixed by @pilu PR and will require PR on status-react to use it, this will allow us to greatly reduce the number of requests when the user have many chats (only 1 request for n chats per day instead of n). **release blocker**
- mailservers that are far away are timing out a lot. For America, Europe and Asia this is less problematic because they can luckily find their local server or manually select it, but other part of the world will have terrible history retrieval performances. My guess is that there is some kind of acknowledgments and waiting between messages sent so the higher the ping the more likely the request is to timeout before all the messages have been sent. This is a **release blocker** IMO. We need to find why the mailserver is so slow at sending messages, if it waits for some kind of acknowledgement or something.
- desktop is not handling lots of loaded message well, this is very visible when fetching seven days of history as my last PR does (better to restart after the week of history has been fetched because the app becomes very irresponsible). This can also be reproduced by scrolling up in a few chats that have a lot of history (I mean scrolling up to the first messages which will take quite some time). We should also try to find a fix for that asap but I don't think it is a release blocker.

**Comment by gravityblast** [comment]
>  The problem with that approach is that there is now a timeout on requests and after that timeout expires, the mailserver stops sending messages (to confirm @pilu @adambabik).

@yenda the timeout is only a client-side timer in status-go. after 10 seconds it's fired if the client hasn't receive the response yet.
The mail server itself never times out. It receives the request, it sends messages 1 by 1, and at the end it sends the "response" message.
This means that even if status-go fires the timeout, the server can continue to send messages.

**Comment by lukaszfryc** [comment]
> we can only request one topic per request (soon fixed by @pilu PR and will require PR on status-react to use it, this will allow us to greatly reduce the number of requests when the user have many chats (only 1 request for n chats per day instead of n). release blocker

@yenda the PR in status-go is already merged. I added an issue for you to use it in status-react https://github.com/status-im/status-react/issues/6334

**Comment by lukaszfryc** [comment]
> mailservers that are far away are timing out a lot. For America, Europe and Asia this is less problematic because they can luckily find their local server or manually select it, but other part of the world will have terrible history retrieval performances. My guess is that there is some kind of acknowledgments and waiting between messages sent so the higher the ping the more likely the request is to timeout before all the messages have been sent. This is a release blocker IMO. We need to find why the mailserver is so slow at sending messages, if it waits for some kind of acknowledgement or something.

@yenda was it always like this or we introduced it somehow recently? If the upcoming release does not introduce regressions in this area I'd say it's not a release blocker. Anyway, it's worth to fix asap.

**Comment by chadyj** [comment]
Team, with the [recent mail server issues](https://github.com/status-im/status-react/issues/6311#issuecomment-430016229) and reports of reliability problems we will need to delay this release until we are absolutely certain Status is working reliably. 

At the Hackathon and Devcon we are showcasing Status to the wider community so performance has to be flawless, and we should take the time to ensure this.

**Comment by lukaszfryc** [comment]
@chadyj is it final decision? How about registration dapp and other things mentioned by @rachelhamlin (https://github.com/status-im/status-react/issues/6311#issuecomment-429775679)? I suppose we can cherry pick at least some of them if we won't make a normal release.

**Comment by chadyj** [comment]
Hey @lukaszfryc it wasn't a decision but an observation. There are several [existing release issues](https://github.com/status-im/status-react/labels/release) that are still open, plus the new mail server issues. Although if there are ways to remedy the issues, salvage the release and have a quality build for hackathon/devcon then lets do it.

**Comment by lukaszfryc** [comment]
I'd say let's aim for the standard release but with a 1 or 2 days delay, submitting the iOS build on Monday.

Backup plan: cherry pick and release only couple of changes like the registration app to support the hackaton. But, I'm not sure which PRs we will be able to cherry pick as many of them may be dependent on on things that are already in develop branch.

----

cc @chadyj @annadanchenko @rachelhamlin @andytudhope.

**Comment by rachelhamlin** [comment]
Makes sense.

In the event that we cherry-pick, I learned there's one more item that marketing needs:
- Universal link issue for QR codes (needed for Incubate/Hackathon): https://github.com/status-im/status-react/issues/6072

And this item for ENS support is not listed in the must-haves above, but would also need to be included:
- Support for all .eth names: https://github.com/status-im/status-react/pull/6279

**Comment by chadyj** [comment]
I propose that we remove Instabug from this release too. It doesn't pass the test as essential software and can expose us to security or privacy incidents. I added the release tag but if anyone feels we should keep it then please holla https://github.com/status-im/status-react/issues/6346

**Comment by chadyj** [comment]
Update: 

We have [3 PR's left](https://github.com/status-im/status-react/pulls?q=is%3Aopen+label%3Arelease+is%3Apr). 

Everything is assigned and in progress.

If all goes well we can cut the release tomorrow and start testing.

**Comment by chadyj** [comment]
Yenda:
> mailservers that are far away are timing out a lot. For America, Europe and Asia this is less problematic because they can luckily find their local server or manually select it, but other part of the world will have terrible history retrieval performances. My guess is that there is some kind of acknowledgments and waiting between messages sent so the higher the ping the more likely the request is to timeout before all the messages have been sent. This is a release blocker IMO.

Pilu:
> the timeout is only a client-side timer in status-go. after 10 seconds it's fired if the client hasn't receive the response yet.
The mail server itself never times out. It receives the request, it sends messages 1 by 1, and at the end it sends the "response" message.
This means that even if status-go fires the timeout, the server can continue to send messages.

@yenda @pilu Is there action required for this? Next steps?

**Comment by goranjovic** [comment]
The remaining PR for sidechains is here #6388

**Comment by mandrigin** [comment]
Just a small note that it is so much easier to track everything here than in Slack.

@chadyj mailserver timing out doesn't mean that you won't receive all the messages anyway, because, as I understand, it is not a p2p timeout, it is our waiting time to receive all the messages.

**Comment by gravityblast** [comment]
@chadyj  we already fixed these things in the last days in `status-go`. Now the mail server sends back an error message, and envelopes can be packed in single messages. The client-side timeout is still needed though in case the mailserver doesn't send the response or in case of connectivity problems

**Comment by mandrigin** [comment]
I think we need to calculate timeout starting from the _last message received from the mailserver_ and not from the actual request time. Maybe we also might want to show how many messages are received.

**Comment by annadanchenko** [comment]
looks like more extensions PR will need to be added according to @jeluard : 
> jeluard [10:12]
We are still missing some bits for extensions otherwise they are essentially unusable..
Also need to push one PR to add kickback to the DApp list

**Comment by chadyj** [comment]
@jeluard @rachelhamlin Help us out and tag any essentials with the `release` tag as well as ping us in this thread.

**Comment by chadyj** [comment]
Have another release issue https://github.com/status-im/status-react/issues/6402
Can someone please grab that?

**Comment by jeluard** [comment]
@chadyj Already tagged, will tag more. I'll let you know once everything is out

**Comment by jeluard** [comment]
@chadyj @alwx is on #6402

**Comment by chadyj** [comment]
Heads up everyone!
There are several new PR's today. Appreciate any help in testing/reviewing/merging 🍻

**Comment by chadyj** [comment]
Release has been cut so testing can commence. Here is what we've got https://github.com/status-im/status-react/compare/release/0.9.29...release/0.9.30

Builds for testing are building here https://jenkins.status.im/job/status-react/job/release/job/release%252F0.9.30/

We will cherry pick the other PR's when done.

**Comment by annadanchenko** [comment]
@jakubgs @rasom all existing 5 builds in https://jenkins.status.im/job/status-react/job/release/job/release%252F0.9.30/ failed. Are you fixing it?

<blockquote><img src="/static/ff3fc112/favicon.ico" width="48" align="right"><div><strong><a href="https://jenkins.status.im/job/status-react/job/release/job/release%252F0.9.30/">release/0.9.30 [status-react » release] [Jenkins]</a></strong></div></blockquote>

**Comment by lukaszfryc** [comment]
FYI, battery consumption looks similar to what it was on latest release (0.9.29). At least on Android device we always used to measure it (Samsung Galaxy S8). ~0.42% (avg of 2 experiments) battery was consumed in a 15 min test. Previously it was ~0.41%. 

<img src="https://user-images.githubusercontent.com/7532782/47223164-d2454b80-d3b8-11e8-93a7-c01dd211ac91.png" width="400">

All results are [here](https://docs.google.com/spreadsheets/d/1G-iKNIet8uoUsJMbxAIz6yYqHCIL3HK6FcdcCfRv2AE/edit#gid=0).

**Comment by jeluard** [comment]
I cherry-picked the relevant PRs we merge after the cut.
We are expecting another PR for extensions.

**Comment by rachelhamlin** [comment]
@jeluard great. what PR is left?

**Comment by jeluard** [comment]
@rachelhamlin A PR restricting the number of extensions installable to 1

**Comment by rachelhamlin** [comment]
@jakubgs @rasom any update on jenkins issues?

**Comment by annadanchenko** [comment]
We can get to the android/ios builds here https://jenkins.status.im/job/status-react/job/release/job/release%252F0.9.30/8/ so not a blocker, but we still have blocking issue with https://jenkins.status.im/job/status-react/job/upload_release_ios/ that uploads iOS build to TestFlight because we need it for testing iOS upgrades cc @jakubgs @rasom @mandrigin 

<blockquote><img src="/static/ff3fc112/favicon.ico" width="48" align="right"><div><strong><a href="https://jenkins.status.im/job/status-react/job/release/job/release%252F0.9.30/8/">status-react » release » release/0.9.30 #8 [Jenkins]</a></strong></div></blockquote>
<blockquote><img src="/static/ff3fc112/favicon.ico" width="48" align="right"><div><strong><a href="https://jenkins.status.im/job/status-react/job/upload_release_ios/">upload_release_ios [status-react] [Jenkins]</a></strong></div></blockquote>

**Comment by mandrigin** [comment]
There is a commit in the previous release branch that fixes upload iOS job. I think I never backported it to develop :(

**Comment by rasom** [comment]
https://gist.github.com/rasom/f0fdf66339184b3ff3fc113edc86ec73 well we also have some slack related error there, probably fixed, checking now

<blockquote><img src="https://avatars2.githubusercontent.com/u/2364994?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> Gist</div><div><strong><a href="https://gist.github.com/rasom/f0fdf66339184b3ff3fc113edc86ec73">release_build_failure</a></strong></div><div>GitHub Gist: instantly share code, notes, and snippets.</div></blockquote>

**Comment by chadyj** [comment]
Ha that is sorta my fault. When I quit Slack it must have broke the Fastlane<>Slack integration which is throwing errors.

We need to remove the Slack actions from the Fastfile like:

```
    slack(
        message: "New release build uploaded to TestFlight",
        slack_url: ENV["SLACK_URL"],
        default_payloads: []
    )
```

**Comment by jeluard** [comment]
Last PR we would like to be part of the release: https://github.com/status-im/status-react/pull/6428

**Comment by annadanchenko** [comment]
Current state (Sunday evening):
1. 2 release PRs still need fixes: https://github.com/status-im/status-react/pull/6388 and https://github.com/status-im/status-react/pull/6428 
2. We can't upload iOS version to TestFlight - https://jenkins.status.im/job/status-react/job/upload_release_ios/ needs fixing or some other way to upload. The faster we submit it to review the better. Next release build with final changes will need less time to review if we got approval of preliminary build.
3. We haven't completed regression testing of the available release build, so need to continue with it on Monday

<blockquote><img src="/static/ff3fc112/favicon.ico" width="48" align="right"><div><strong><a href="https://jenkins.status.im/job/status-react/job/upload_release_ios/">upload_release_ios [status-react] [Jenkins]</a></strong></div></blockquote>

**Comment by mandrigin** [comment]
I'll take a look at (2) tomorrow, Apple still doesn't look at the test builds on weekends, so the earliest review we can get is tomorrow evening (European time) anyway.

**Comment by rachelhamlin** [comment]
Alex is looking at #6428 this morning.

**Comment by mandrigin** [comment]
Will test uploading to TestFlight now.

**Comment by mandrigin** [comment]
Will test uploading to TestFlight now.

**Comment by mandrigin** [comment]
testing uploading to TestFlight

**Comment by mandrigin** [comment]
`upload_release_ios` is fixed on 0.9.30, will fix the `develop` now.

**Comment by chadyj** [comment]
Thanks @mandrigin.  I got the TestFlight build and was able to install.

Still have 2 PR's in progress from @goranjovic @alwx.

Heads up that there is also a suggestion to add PR's https://github.com/status-im/status-react/pull/6425 and https://github.com/status-im/status-react/pull/6448 to the release. I added the release tags @rasom can you please take a look and see if we should include it?

Good luck with the regression testing, testteam!

**Comment by jeluard** [comment]
#6428 has been tested, merged and cherry-picked

**Comment by Serhy** [comment]
https://github.com/status-im/status-react/issues/6460 - one more regression we need to include in 0.9.30 release

**Comment by rachelhamlin** [comment]
Alex looking at #6428 this morning.

**Comment by chadyj** [comment]
@mandrigin If we want this to be ready for the hackathon then when is the absolute latest we should submit to Apple?  We are getting really close, and I expect much of the team will be travelling so we should set a hard deadline or we risk being late.

@annadanchenko How did regression testing go?

**Comment by mandrigin** [comment]
@chadyj we have an older build of 0.9.30 approved (I submitted it yesterday), so we theoretically should get Insta-approvals for the newer builds too. In the meantime, can you post release notes here?

**Comment by chadyj** [comment]
Thanks Igor. 

Release notes are here https://docs.google.com/document/d/1Z0mFlJJkNbRgv6qm80ob5g4PEvsI8cTmbwZpcRPWPho/edit

<blockquote><img src="https://lh5.googleusercontent.com/8HPrqui70EEzEnxev4gaBEgitMo_UkXOvajQPFY64N_kB8XBJvsdAZkxd61qeQCb7xNTig=w1200-h630-p" width="48" align="right"><div>Google Docs</div><div><strong><a href="https://docs.google.com/document/d/1Z0mFlJJkNbRgv6qm80ob5g4PEvsI8cTmbwZpcRPWPho/edit?usp=embed_facebook">Release Notes - Ongoing</a></strong></div><div>Process for status releases:  Each status release needs a brief 3 - 5 paragraph story to tell about the successes, challenges and features within the release. Work with #core-releases to compare commits to facts of “New & Improved and Bug Fixes” section - Commit differences can be found here and ...</div></blockquote>

**Comment by lukaszfryc** [comment]
Re regression testing, we have not done the full regression yet because of [open issues](https://github.com/status-im/status-react/labels/release). We have 2 open PRs that may introduce regressions in messaging area: https://github.com/status-im/status-react/pull/6466 and https://github.com/status-im/status-react/pull/6448. And https://github.com/status-im/status-react/pull/6388 that may affect transactions. I'll start testing 6466 and 6448 while they are still in review.

@chadyj I think we absolutely need to close them today to submit the new build tomorrow.

**Comment by jeluard** [comment]
I had to cherry-pick a tiny tiny extension bug: #6463

**Comment by annadanchenko** [comment]
@jeluard I see you marked it (#6463) with skip-qa and merged directly to release. Can you comment on any tests done by yourself for it? will you check it also in the release build?

**Comment by jeluard** [comment]
@annadanchenko Yes this fix relates to changes needed to have some specific feature of extension working. Harmless out of the scope of extension. Will validate manually myself.

**Comment by annadanchenko** [comment]
>@annadanchenko How did regression testing go?

@chadyj on yesterday's release build test team completed most of critical and high priority tests and found no more blockers besides known in https://github.com/status-im/status-react/issues?q=is%3Aopen+is%3Aissue+label%3Arelease 

Unfortunately, we are still in progress of fixing 2 issues for duplicated messages after upgrade (#6460, #6438) and an issue to add new networks #6250 that also impacts upgrade. So as soon as they are done, we will need to run regression tests again and it can bring more surprises. 

<blockquote><img src="https://avatars3.githubusercontent.com/u/11767950?s=400&v=4" width="48" align="right"><div><img src="https://assets-cdn.github.com/favicon.ico" height="14"> GitHub</div><div><strong><a href="https://github.com/status-im/status-react">status-im/status-react</a></strong></div><div>a free (libre) open source, mobile OS for Ethereum - status-im/status-react</div></blockquote>

**Comment by jeluard** [comment]
How mandatory is #6250 ?
We must have the new release out for the hackathon or we can't test easily extensions.

**Comment by annadanchenko** [comment]
> How mandatory is #6250 ?

imo, if we can't fix it today we need to drop it to release the rest.

but there are upgrade issue and they are not pretty at all to release with, @janherich any estimates when fixes will be ready?

**Comment by rachelhamlin** [comment]
I would also vote to drop it in order to get us closer to release (cc @andytudhope)

**Comment by asemiankevich** [comment]
@chadyj ^

**Comment by annadanchenko** [comment]
@yenda added https://github.com/status-im/status-react/pull/6478 to the release too. However, I'm not sure how critical it is (described scenario works fine without this PR or steps are wrong in the PR), waiting for Eric to comment

**Comment by goranjovic** [comment]
regarding #6250 - we can bypass the upgrade decryption problem if we don't include the migration to add two sidechains to the existing accounts. In practice, this would mean that hackaton participants who want to use the sidechains would have to use a new account (but at least they would be able to do that)

wdyt @annadanchenko @andytudhope ?

**Comment by chadyj** [comment]
> regarding #6250 - we can bypass the upgrade decryption problem if we don't include the migration to add two sidechains to the existing accounts. In practice, this would mean that hackaton participants who want to use the sidechains would have to use a new account.

Sounds like a decent trade off to get this release shipped.

**Comment by goranjovic** [comment]
@chadyj Ok, I'll remove the migrations then.

**Comment by yenda** [comment]
@goranjovic maybe there is a bug in the migration ?

**Comment by rachelhamlin** [comment]
So this PR should solve both #6460 + #6438: https://github.com/status-im/status-react/pull/6488

But build process is delaying us, per `Ragged Understated Loon`. 😕

**Comment by pablanopete** [comment]
> regarding #6250 - we can bypass the upgrade decryption problem if we don't include the migration to add two sidechains to the existing accounts. In practice, this would mean that hackaton participants who want to use the sidechains would have to use a new account (but at least they would be able to do that)
> 
> wdyt @annadanchenko @andytudhope ?

also agree this won't be a problem because xDaichain is a brand new side chain so people won't have any tokens in it before the hackathon anyways very likely - creating a new account is a bit of a UX slowdown but also agree to get it in release shipped and since it's blocking old accounts it makes sense.

**Comment by annadanchenko** [comment]
@rachelhamlin is there anything to be changed in the mobile app for ENS usernames? like linking to the Dapp url on mainnet ?

**Comment by chadyj** [comment]
We are 2 days away from the Hackathon and so really have to wrap up this release today.  People will also be traveling so availability will be limited.

The de-duplication work is being done so thanks @janherich @rasom for making it a priority. Please yell out if you are blocked or can't get this done in a few hours.

@goranjovic do you expect to wrap up the sidechain issue this morning, or should we drop it from the release? 

@rachelhamlin @jeluard are things all set for extensions? 
When is ENS launching on Mainnet? AFAIK it still needs to be tested in the release

@j-zerah @kimjf Are you all set with release notes?

If things aren't in a good spot by 12CEST we should jump on a call and discuss how to proceed.

**Comment by annadanchenko** [comment]
@goranjovic @chadyj despite known non-critical issues I propose to merge #6388 and document known issues in release notes. see them in https://github.com/status-im/status-react/pull/6388#issuecomment-432549313

**Comment by jeluard** [comment]
@chadyj Everything is ready for extensions.

Let's :ship: :ship: :ship:

**Comment by rachelhamlin** [comment]
@chadyj @annadanchenko nothing to change in status-react. The entry point on the profile screen links to `names.statusnet.eth`. 

The ENS _contract_ still needs to be deployed to mainnet, which will require a change with the `names.statusnet.eth` ENS resolver, but nothing more to do in Status to support this.

We're aiming to have this done today, so it will happen ahead of the release.

**Comment by rachelhamlin** [comment]
We've tested the ENS DApp thoroughly on Ropsten and as soon @bgits or @3esmit deploys the contract to mainnet, will test there as well.

**Comment by chadyj** [comment]
⚡️⚡️⚡️All the outstanding `release` issues have been closed. ⚡️⚡️⚡️

The release build [can be downloaded from jenkins](https://ci.status.im/job/status-react/job/release/job/release%252F0.9.30/24/) for testing.

However the Testflight upload [failed](https://ci.status.im/job/status-react/job/upload_release_ios/52/console) complaining about `Invalid username and password combination`. @mandrigin @jakubgs Can you please take a look?

**Comment by chadyj** [comment]
Roman fixed (thanks) the Fastlane issue and build 11274 is ready for testing in Testflight. ⚡️⚡️⚡️

**Comment by kimjf** [comment]
I am traveling today but should be able to help with the release tomorrow. Do we need an intro written for the notes? If we do could you highlight in the notes the key updates to call out?

**Comment by chadyj** [comment]
I noticed one issue in the TF version where messages are not ordered properly. Older messages are shown before newer messages.

<img width="57" alt="screen shot 2018-10-24 at 2 26 44 pm" src="https://user-images.githubusercontent.com/116099/47430204-0940a600-d799-11e8-90b1-43341a4e0baf.png">

@yenda Fixed a related ordering issue today on desktop. Same cause?

**Comment by jeluard** [comment]
@chadyj I have this issues for days. Probably not a recent regression.

**Comment by rachelhamlin** [comment]
@kimjf no stress, already written! Coordinating with Jonny on the posts.

**Comment by annadanchenko** [comment]
test team is very close to complete regression testing on the last release build. So far, no blocking issues found. In ~30 min we expect to complete testing and potentially we are OK to release.

@chadyj when do we want to release it? On TestFlight it's 11274 in 0.9.30. For Android: https://ci.status.im/job/status-react/job/release/job/release%252F0.9.30/24/artifact/pkg/StatusIm-181024-080818-46d265-release.apk cc @mandrigin 

<blockquote></blockquote>

**Comment by yenda** [comment]
@chadyj it is not the same cause because the cause on desktop wasn't related to mobile at all. I think it is because the timestamp of chats isn't based on last message but time when the last message was received in the chat, so when receiving messages from inbox the chats are ordered based on when the last message was received from there

**Comment by chadyj** [comment]
Nice one @annadanchenko and team!

If everything looks good let's shoot for 10am CEST tomorrow! 🌮 

We will use 11274 in TF and 11274 in Google Play currently in the internal test track.

**Comment by rachelhamlin** [comment]
Didn't realize this was already approved by Apple! Nice job, everyone!

Might be preferable to release later in the day, from a product perspective.

We just deployed the ENS contract to mainnet, and Barry is making a few last changes to the DApp before we can test on mainnet. 

Once `0.9.30` is live & announced, people will know they can register names. @asemiankevich and I will test the DApp as soon as the latest is deployed today, but more buffer might be good. We were initially thinking we'd announce around 9 ET tomorrow. 

@j-zerah thoughts from a marketing perspective? There is a blog post for both `0.9.30` and for `ENS`, and Blonde is also making a PR about ENS.

**Comment by rachelhamlin** [comment]
Update: [ENS DApp](https://cloudflare-ipfs.com/ipfs/QmSKe8h1QJ2h2QWWqCC3nwVSA4ekWgKZ7MBsTDxwP4Hchf/) works well in my run-through of each core flow, but there's one tiny and fatal issue. The price is wrong below the fold on the first screen. 

This requires attention from both @bgits and @3esmit to re-deploy the DApp, once fixed. After speaking with @j-zerah, think it makes sense to target 3 pm CEST to release 0.9.30.

@annadanchenko @chadyj

**Comment by annadanchenko** [comment]
@chadyj @mandrigin 11274 build is in Ready to Submit state that looks like it wasn't submitted to review. Can you send it to review if it's so, please?

**Comment by annadanchenko** [comment]
nevermind, just did it myself and got approval right after this

**Comment by mandrigin** [comment]
done. approved. "testing"

**Comment by rachelhamlin** [comment]
Latest version of the ENS DApp is deployed, now we need Ricardo to update the ENS URI to contain the new IPFS hash.

He should be landing in Prague soon, but I think we should push the release back one more hour to 4 PM CEST to be safe.

**Comment by chadyj** [comment]
0.9.30 is released! 

https://our.status.im/v0-9-30-release-extensions-ens-security/

<blockquote><img src="https://our.status.im/content/images/2018/10/Wallet_upgrade__0.9.30.jpg" width="48" align="right"><div>Our Status</div><div><strong><a href="https://our.status.im/v0-9-30-release-extensions-ens-security/">v0.9.30 Release – Extensions, ENS Names, and more!</a></strong></div><div>The Status Hackathon - #CryptoLife is taking place in Prague this weekend (October 26 - 28), so this release goes out to the developers who will be hacking with us both at the event and around the world.ExtensionsBuilding a DApp? You can now extend Status functionality to make your</div></blockquote>
